### ISSUE TITLE
JCoreDic

### ISSUE THREAD

**Issue by heliosroots** [jcmoraisjr/jcore#7]
TJCoreDIC does not call the procedure view in the constructor?

http://pastebin.com/YB1Citiv

**Comment by jcmoraisjr** [comment]
You should call the intf.view method:

    var
      MyInterface: IMyInterface;
    begin
      // register an implementation
      TJCoreDIC.Register(IMyInterface, 'TMyObject', TMyObject, jdsRequest);
      // locate an implementation
      TJCoreDIC.Locate(IMyInterface, 'TMyObject', MyInterface);
      // use the implementation located by DIC
      MyInterface.View;
    end.

**Comment by heliosroots** [comment]
yes I understand the operation, so I must forget what is inside the constructor of my concrete classes? if so is not a good behavior?

**Comment by jcmoraisjr** [comment]
Ah, now I see the View being called inside the constructor. The problem is that TObject constructors are not virtual nor published, so DIC cannot call it. I will leave this issue open until I document this somewhere.

The only way I can think you can trigger something on the class instantiation is overriding AfterConstruction method.

**Comment by heliosroots** [comment]
Yes, it is feasible to use AfterConstruction! It would be nice to leave a comment on TJCoreDic alert! Thanks.
